index.d.ts

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react'
+
+declare class ReactSwipeable<T extends Element = HTMLElement> extends React.Component<
+  ReactSwipeable.SwipeableProps<T>
+> {}
+
+interface EventData<T> {
+  event: React.TouchEvent<T>;
+  deltaX: number;
+  deltaY: number;
+  absX: number;
+  absY: number;
+  velocity: number;
+  dir: string;
+}
+
+declare namespace ReactSwipeable {
+  type OnSwipingCallback<T extends Element = HTMLElement> = ({}: EventData<T>) => void
+  type OnSwipedCallback<T extends Element = HTMLElement> = ({}: EventData<T>) => void
+  type OnSwipedDirectionCallback<T extends Element = HTMLElement> = (
```

Comment:
all the `onSwipedDirection`'s also use the `EventData`

Reply:
Is `isFlick` deprecated?

Reply:
yup, [migration docs](https://github.com/dogfessional/react-swipeable/blob/master/migration.md#flickthreshold)